### PR TITLE
[XPU] support ep+tp at model layer

### PR DESCRIPTION
**Pull request by zhupengyang**
## Motivation

xpu支持ep+tp

## Modifications
- 修复eb56 moe组网代码。因为tp的token划分是不均匀的，token_num比较小可能导致id靠后的卡索引错误

## Checklist

- [ ] Add at least a tag in the PR title.
  - Tag list: [`[FDConfig]`,`[APIServer]`,`[Engine]`, `[Scheduler]`, `[PD Disaggregation]`, `[Executor]`, `[Graph Optimization]`, `[Speculative Decoding]`, `[RL]`, `[Models]`, `[Quantization]`, `[Loader]`, `[OP]`, `[KVCache]`, `[DataProcessor]`, `[BugFix]`, `[Docs]`, `[CI]`, `[Optimization]`, `[Feature]`, `[Benchmark]`, `[Others]`, `[XPU]`, `[HPU]`, `[GCU]`, `[DCU]`, `[Iluvatar]`, `[Metax]`]
  - You can add new tags based on the PR content, but the semantics must be clear.
- [ ] Format your code, run `pre-commit` before commit.
- [ ] Add unit tests. Please write the reason in this PR if no unit tests.
- [ ] Provide accuracy results.
- [ ] If the current PR is submitting to the `release` branch, make sure the PR has been submitted to the `develop` branch, then cherry-pick it to the `release` branch with the `[Cherry-Pick]` PR tag.

### CODE REVIEW

File: scripts/run_ci_xpu.sh [hunk 1]
```diff
@@ -201,14 +205,47 @@ unset BKCL_PCIE_RING
...
+    echo "EP4TP1 相关测试失败，请检查pr代码"
+    exit 1
+fi
+
+echo "============================开始 EP4TP4 测试!============================"
+sleep 5
+rm -rf log/*
+rm -f core*
+xpu-smi
+export XPU_VISIBLE_DEVICES="0,1,2,3"
```
Comment (iosmers) [inline scripts/run_ci_xpu.sh:222]: 这些是跑EP必须开的环境变量吗

File: fastdeploy/distributed/communication.py [hunk 2]
```diff
@@ -56,6 +56,9 @@ def tensor_model_parallel_all_reduce(
         group_: paddle.distributed.communication.group.Group = None,
     ) -> paddle.Tensor:
         """All-reduce the input tensor across model parallel group."""
+        if input_.numel() == 0:
```
Comment (iosmers) [inline fastdeploy/distributed/communication.py]: 这里也是处理EP空跑的情况吗

Comment (zhupengyang) [reply fastdeploy/distributed/communication.py]: 是的

File: scripts/run_ci_xpu.sh (hunk 1 above)
Comment (zhupengyang) [reply scripts/run_ci_xpu.sh:222]: 是的，ep通信必须开

File: fastdeploy/model_executor/layers/embeddings.py [hunk 3]
```diff
@@ -301,6 +302,13 @@ def forward(self, ids_remove_padding=None) -> paddle.Tensor:
             )
             input_embedings = paddle.concat(inputs_embeds_temp, -1)
         else:
+            if current_platform.is_xpu():
+                token_num = ids_remove_padding.numel()
+                if token_num <= 0:
+                    embedding_dim = self.embeddings.weight.shape[1]
+                    dtype = self.embeddings.weight.dtype
+                    input_embedings = paddle.empty((token_num, embedding_dim), dtype=dtype)
+                    return input_embedings
```
Comment (zhupengyang) [inline fastdeploy/model_executor/layers/embeddings.py]: cudagraph不支持获取运行时数据做if判断，所以只在xpu下做token=0的跳过

File: fastdeploy/model_executor/models/ernie4_5_moe.py [hunk 4]
```diff
@@ -224,19 +224,19 @@ def update_state_dict(self, state_dict):
...
         part_hidden_states[: (end_offset - start_offset), :] = hidden_states[start_offset:end_offset, :]
         out = self.experts(part_hidden_states, self.gate)
-        multi_outs = []
+        multi_outs = paddle.empty(
+            [token_num_per_rank * self.tensor_parallel_size, hidden_states.shape[1]], dtype=hidden_states.dtype
+        )
         paddle.distributed.all_gather(multi_outs, out, self.tp_group)
-        out = paddle.concat(multi_outs, axis=0)
-        out = out[:token_num, :]
+        out = multi_outs[:token_num, :]
```
Comment (yuanlehome) [inline fastdeploy/model_executor/models/ernie4_5_moe.py:243]: 这一块我推荐不做这么多改动，多加下面一个判断就可以work了
```c++
if end_offset > token_num:
    end_offset = token_num
if start_offset > end_offset:
    start_offset = end_offset
```

File: fastdeploy/model_executor/layers/embeddings.py (hunk 3 above)
Comment (yuanlehome) [reply fastdeploy/model_executor/layers/embeddings.py]: 空转的逻辑，参考下gpu_model_runner的_execute_empty_input函数，只空转moe即可

File: fastdeploy/model_executor/models/ernie4_5_moe.py (hunk 4 above)
Comment (zhupengyang) [reply fastdeploy/model_executor/models/ernie4_5_moe.py:243]: start_offset是bug修复，可以按照你说的改。

zeros改empty，all_gather的输出list改tensor是为了性能优化，xpu上验证过是有效的

File: fastdeploy/model_executor/layers/embeddings.py (hunk 3 above)
Comment (zhupengyang) [reply fastdeploy/model_executor/layers/embeddings.py]: done

**Comment by yuanlehome** [comment]
qwen3_moe里也有split_allgather_out，是不是一并改了，最好这个逻辑是下放到FusedMoE layer里去，这样所有moe模型都能用
